### ISSUE TITLE
*** No instances to display *** 

### ISSUE THREAD

**Issue by leventozdemir** [matterport/Mask_RCNN#2132]
i have a data of 1000 labelled image using VGG Image Annotator (900 train , 100 val), the problem when the loss of train get down 0.9* i start to get an over fitting problem.
i kept training it until loss = 0.006* then i test the model using images from training set but i get the the out without any bounding box and without any mask  (*** No instances to display *** ) .
i used 'coco' and i used 'imageNet' pre trained weights , also i changed step per epoch and the i changed the learning rate and i used ResNet101 and RestNet50 and i still face the problem .
i think the problem in my data . 
any one can tell what should my data type must and what should the size of the images and the best way to label them ?

**Comment by ejri** [comment]
Hey, so you could try a few things. 

- Lower the detection threshold. 
- use scipy versions less than 1.3.0
- There was another thread about some possible issues due data type error because of float numbers becoming zero

modify utils.py line 866 from
shift = np.array([0, 0, 1, 1])
to
shift = np.array([0, 0, 1., 1.])

(https://github.com/matterport/Mask_RCNN/issues/912)

**Comment by leventozdemir** [comment]
> Hey, so you could try a few things.
> 
> * Lower the detection threshold.
> * use scipy versions less than 1.3.0
> * There was another thread about some possible issues due data type error because of float numbers becoming zero
> 
> modify utils.py line 866 from
> shift = np.array([0, 0, 1, 1])
> to
> shift = np.array([0, 0, 1., 1.])
> 
> (#912)

Thank you for answer me but i triedall the things above expect (Lower the detection threshold.)
How can i try this ??

**Comment by ejri** [comment]
it goes in your training config:

    # Minimum probability value to accept a detected instance
    # ROIs below this threshold are skipped
    DETECTION_MIN_CONFIDENCE = 0.90

    # Non-maximum suppression threshold for detection
    DETECTION_NMS_THRESHOLD = 0.3


try lowering both

**Comment by leventozdemir** [comment]
thank you very much i'll try it .

**Comment by leventozdemir** [comment]
> it goes in your training config:
> 
> ```
> # Minimum probability value to accept a detected instance
> # ROIs below this threshold are skipped
> DETECTION_MIN_CONFIDENCE = 0.90
> 
> # Non-maximum suppression threshold for detection
> DETECTION_NMS_THRESHOLD = 0.3
> ```
> 
> try lowering both

thank you, this work for me and i have another question if you have any article or any thing which explain this two variables can you share it with me .

**Comment by ejri** [comment]
Honestly, by googling, you'll find a lot more useful information than I can provide.

DETECTION_MIN_CONFIDENCE = 0.90

is the minimum threshold of confidence you give for a prediction to be detected. For example: you have a model to detect cats in images. If you tell your model to predict a test image, and the model tells you that the image is a cat with 80%  confidence, the model will ignore it because it's less than the 90% threshold you gave it. 

Non-maximum suppression threshold for detection
DETECTION_NMS_THRESHOLD = 0.3

I'm not exactly sure what it does, but if your model is detecting the same cat, in the same test image, with multiple and overlapping bounding boxes around the cat, lowering this threshold helps. 

[check this article](https://towardsdatascience.com/pedestrian-detection-using-non-maximum-suppression-b55b89cefc6)

**Comment by leventozdemir** [comment]
> Honestly, by googling, you'll find a lot more useful information than I can provide.
> 
> DETECTION_MIN_CONFIDENCE = 0.90
> 
> is the minimum threshold of confidence you give for a prediction to be detected. For example: you have a model to detect cats in images. If you tell your model to predict a test image, and the model tells you that the image is a cat with 80% confidence, the model will ignore it because it's less than the 90% threshold you gave it.
> 
> Non-maximum suppression threshold for detection
> DETECTION_NMS_THRESHOLD = 0.3
> 
> I'm not exactly sure what it does, but if your model is detecting the same cat, in the same test image, with multiple and overlapping bounding boxes around the cat, lowering this threshold helps.
> 
> [check this article](https://towardsdatascience.com/pedestrian-detection-using-non-maximum-suppression-b55b89cefc6)

thank very much .
